### ISSUE TITLE
Documentation needed

### ISSUE THREAD

**Issue by bhilbert4** [spacetelescope/nircam_simulator#17]
Need to write a technical report in order to fully describe the simulator (also be sure to include ancillary useful code, like the yaml_generator). 

TR is in progress. Needs lots of updates from when I started writing it months ago to describe a much less mature code base.

**Comment by bhilbert4** [comment]
TR is essentially done. Waiting for IT to get back to me on where to host the repository of config/reference files.

**Comment by bhilbert4** [comment]
TR submitted to SOCCER. Comments from independent review should come back relatively soon.

Also to do: proper (PEP8, PEP256) documentation for each function.

**Comment by bhilbert4** [comment]
I think example notebooks are the best place to start. I'll also send you a copy of the "handbook" which is in TR form in the middle of the SOCCER review process at the moment.

**Comment by laurenmarietta** [comment]
When I was teaching myself how MIRAGE worked, I tried to develop a kind of "outline" that lists everything that happens when the simulator runs. 

Attaching it here; maybe it (or some variation of it) could be useful to include in the documentation somewhere:
[RampSimulatorOutline.docx](https://github.com/spacetelescope/mirage/files/2294062/RampSimulatorOutline.docx)

**Comment by bhilbert4** [comment]
That's true. I need to learn Sphinx so that `mirage` can have better documentation. The NIRCam team has also been pushing for me to turn the TR into a PASP paper in order to make it more accessible (and to make a better advertisement).

**Comment by mperrin** [comment]
Sphinx is super easy. I'd be happy to set up the basic framework in a PR, and then you can paste in content into that over time.

**Comment by mperrin** [comment]
See #147
